### PR TITLE
Avoid using array index as `key` prop

### PR DESCRIPTION
**Pull request by ascott**
plz review @alvinsng @spikebrehm

### CODE REVIEW

File: react/README.md [hunk 1]
```diff
@@ -279,6 +279,22 @@
   <div />
   ```
 
+- Avoid using array index as `key` prop. ([why?](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318))
```
Comment (lencioni) [inline react/README.md]: What do you think about phrasing this more generally, in a way that makes it clear what kinds of keys are good to use?

File: react/README.md [hunk 2]
```diff
@@ -279,6 +279,22 @@
...
+
+```jsx
```
Comment (spikebrehm) [inline react/README.md]: The alignment is off compared to the other items in the list. I think the way they keep it indented is to add a `> ` block before the code sample.

screen shot 2016-05-25 at 12 11 29 pm

File: react/README.md [hunk 3]
```diff
@@ -279,6 +279,22 @@
...
+// bad
+{todos.map((todo, index) =>
+  <Todo {...todo}
```
Comment (spikebrehm) [inline react/README.md]: Let's have these examples conform to our prop formatting rules:

```jsx
// bad
{todos.map((todo, index => 
  <Todo
    {...todo}
    key={index}
  />
}
```

**Comment by spikebrehm** [comment]
Thanks for this! Content LGTM, but there's some formatting suggestions.

File: react/README.md (hunk 2 above)
Comment (lencioni) [inline react/README.md]: I don't think the indentation is related to the `>` blockquote. I think it is just the codeblock needs to be indented.

File: react/README.md (hunk 1 above)
Comment (ljharb) [inline react/README.md]: also, "array indexes" or "array indices"?

Comment (ascott) [inline react/README.md]: yeah good point. what about:

`Avoid using array index as `key` prop, prefer a unique id.`

Comment (ljharb) [inline react/README.md]: `s/id/ID` and sounds great!

Comment (alvinsng) [inline react/README.md]: Use unique element ids and not array indexes or other counter indexes as React may display elements out of order. See this article for why.

**Comment by alvinsng** [comment]
Thanks!

**Comment by ascott** [comment]
fixed the formatting: 
screen shot 2016-05-25 at 12 23 54 pm

**Comment by ljharb** [comment]
@ascott looks great! any chance you'd mind rebasing down to one commit? :-)

**Comment by ascott** [comment]
will do @ljharb

**Comment by goatslacker** [comment]
Can we get a lint rule for this? :)

**Comment by ljharb** [comment]
There's https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-key.md but I don't know how reliably a lint rule would be that checked for "using the second param to a map callback" or "using something called i or index". Please do file an issue on `eslint-plugin-react` if you think one could still be written.

**Pull request merged**